### ISSUE TITLE
Delete using filter leaves objects searchable

### ISSUE THREAD

**Issue by TheBritishSabrina** [semi-technologies/weaviate#1836]
**When deleting by uuid, some objects remain searchable in the schema.**

Steps to reproduce:
- Search for objects using a filtered get query
- Extract object ids from query result
- Delete objects by id
- Search for the same objects using a filtered get query

Code snippet to reproduce:
```word_count = 1000
where_filter = {"path": ["word_count"], "operator": "Equal", "valueInt": word_count}

res = (
    weaviate_helper.client.query.get("Books", ["word_count"])
    .with_additional("id")
    .with_where(where_filter)
    .do()
)

ids_to_delete = []
for example in res["data"]["Get"]["Books"]:
    ids_to_delete.append(example["_additional"]["id"])

for id in ids_to_delete:
    print(weaviate_helper.client.data_object.exists(id))
    weaviate_helper.client.data_object.delete(id)
    print(weaviate_helper.client.data_object.exists(id))
    # prints True - False for each id

weaviate_helper.client.query.get("Books", ["word_count"]).with_additional("vector").with_where({"path": "word_count", "operator": "Equal", "valueInt": 1000}).do() 
# returns all objects that existed before deletion

**Comment by antas-marcin** [comment]
Hello @TheBritishSabrina thank you once again for reporting this issue.

Could you tell us which version of Weaviate you're using?

We have identified that this bug has been fixed in `1.9.1` version, so if you're using a lower version of Weaviate then `1.9.1` then an upgrade of Weaviate to the latest version (our latest release is `1.10.1`) should fix it.

Please let us know if the upgrade fixed your issue.

**Comment by TheBritishSabrina** [comment]
Hi @antas-marcin, thanks for getting back to me. 

That's strange, I'm pretty sure we're using `v1.10.1`, as the docker image is `docker.io/semitechnologies/weaviate:1.10.1`. We haven't set a version of the python client, so it will install the latest one (`v3.2.5`?) when it's deployed to kubernetes. 

When I run the python client from my local machine though, it says the version is `3.3.0`. Is that correct?

**Comment by TheBritishSabrina** [comment]
Hi @antas-marcin, I'm afraid I can't be more specific with respect to weaviate on the cluster as I don't have access to its internal system, but it would be strange for the named image to be a higher version than it's running. 

The python client is currently `v3.3.0` on my machine (where I've tested this issue most) and `v3.3.2` on the cluster, which displays the same issue.

**Comment by antas-marcin** [comment]
@TheBritishSabrina you can also double check Weaviate's version by invoking `get_meta()` method using python's client.

```
meta_info = weaviate_helper.client.get_meta()
print(meta_info['version'])
```

**Comment by antas-marcin** [comment]
Oh, ok so it's `v1.10.1` for sure then and you're able to reproduce this bug. hmm so let me get back to you, I'll try to reproduce it on our side.

**Comment by TheBritishSabrina** [comment]
Ok great, thank you!

**Comment by antas-marcin** [comment]
Hello @TheBritishSabrina 

we have been able to find the cause of this issue.

Your explanation and example provided in this issue was very helpful for us. Thank you for that!

This bug occurs if you have UUIDs which are not unique across classes. So basically if you import the same UUID into Class1 as in Class2 then the delete currently only deletes the first match it finds (which is pretty random according to which Class it checks first).

Make sure that UUIDs are really unique, also across classes. Classes act somewhat like namespaces, but unfortunately not completely because some operations such as DELETE do not require the namespace as arguments, so they have to run across Classes. We will improve this in the long-term and make the APIs respect Class namespaces

We have pushed a fix (PR #1845) that makes sure that a DELETE /v1/objects/{id} really deletes every object with that {id} even if it occurs in multiple classes whereas currently it would just delete the first one it finds.
